src/back/glsl/mod.rs

```diff
@@ -3562,6 +3592,19 @@ impl<'a, W: Write> Writer<'a, W> {
         Ok(())
     }
 
+    /// Issue a memory barrier. Please note that to ensure visibility,
+    /// OpenGL always requires a call to the `barrier()` function after a `memoryBarrier*()`
+    fn write_barrier(&mut self, flags: crate::Barrier, level: back::Level) -> BackendResult {
+        if flags.contains(crate::Barrier::STORAGE) {
+            writeln!(self.out, "{}memoryBarrierBuffer();", level)?;
+        }
+        if flags.contains(crate::Barrier::WORK_GROUP) {
+            writeln!(self.out, "{}memoryBarrierShared();", level)?;
+        }
+        writeln!(self.out, "{}barrier();", level)?;
+        Ok(())
+    }
+
```

Comment:
nitpick(non-blocking): It would have been nice to have this refactored code be put into a separate commit.